external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java

```diff
@@ -137,8 +137,14 @@ public EmitState next(SpoutOutputCollector collector) {
             }
             Iterable<List<Object>> tups = KafkaUtils.generateTuples(_spoutConfig, toEmit.msg);
             if (tups != null) {
-                for (List<Object> tup : tups) {
-                    collector.emit(tup, new KafkaMessageId(_partition, toEmit.offset));
+		if(_spoutConfig.topicAsStreamId) {
+	            for (List<Object> tup : tups) {
```

Comment:
@harshach 
Having the if condition check inside the for loop is a performance overhead.
About renaming the variables, I did not name them. The original storm-kafka creators did. So I am not going to rename them now.